### PR TITLE
qa_crowbarsetup: DRY rubyjsonparse function

### PR DESCRIPTION
**Pull request by bmwiedemann**
we had 12x duplicated ruby json parser code
which is now moved to a single place in the rubyjsonparse function

### CODE REVIEW

**Comment by jdsn** [comment]
:+1: very nice

**Comment by aspiers** [comment]
+1000000000

Although there is a lot more refactoring which could be done on top of that (especially if we were using pure Ruby instead of bash ;-)  Maybe I will suggest some inline.

**Comment by jdsn** [comment]
@aspiers this is just one step. Refactoring will go on.

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -1978,9 +1976,9 @@ function get_novadashboardserver()
 function get_ceph_nodes()
 {
     if [[ -n "$deployceph" ]]; then
-        cephmons=`crowbar ceph proposal show default | $ruby -e "require 'rubygems';require 'json';puts JSON.parse(STDIN.read)['deployment']['ceph']['elements']['ceph-mon']"`
-        cephosds=`crowbar ceph proposal show default | $ruby -e "require 'rubygems';require 'json';puts JSON.parse(STDIN.read)['deployment']['ceph']['elements']['ceph-osd']"`
-        cephradosgws=`crowbar ceph proposal show default | $ruby -e "require 'rubygems';require 'json';puts JSON.parse(STDIN.read)['deployment']['ceph']['elements']['ceph-radosgw']"`
+        cephmons=`crowbar ceph proposal show default | rubyjsonparse "puts j['deployment']['ceph']['elements']['ceph-mon']"`
+        cephosds=`crowbar ceph proposal show default | rubyjsonparse "puts j['deployment']['ceph']['elements']['ceph-osd']"`
+        cephradosgws=`crowbar ceph proposal show default | rubyjsonparse "puts j['deployment']['ceph']['elements']['ceph-radosgw']"`
```
Comment (aspiers) [inline scripts/qa_crowbarsetup.sh]: Would be good to add another commit on top of this (maybe in a new PR when this one is merged) which does further refactoring here, e.g.:

- [ ] Just call `crowbar ceph proposal show default` once, and save the result for reuse.
- [ ] Eliminate the `rubyjsonparse "puts j['deployment']['ceph']['elements']..."` duplication with another shell function

**Comment by aspiers** [comment]
Of course, we have an epic for refactoring ;-)  I just wanted to give some examples of further simple refactorings which can be done immediately.
